src/coreclr/nativeaot/Runtime/eventpipe/ep-rt-aot.cpp

```diff
@@ -834,11 +834,23 @@ ep_rt_thread_handle_t ep_rt_aot_thread_get_handle (void)
     return ThreadStore::GetCurrentThreadIfAvailable();
 }
 
+ep_rt_thread_handle_t ep_rt_aot_thread_get_handle_for_streaming (void)
+{
+    // This is expensive but need a valid thread that will only be used by EventPipe
+    ThreadStore::AttachCurrentThread();
+    return ThreadStore::GetCurrentThread();
+}
+
 ep_rt_thread_id_t ep_rt_aot_thread_get_id (ep_rt_thread_handle_t thread_handle)
 {
     return thread_handle->GetPalThreadIdForLogging();
 }
 
+bool ep_rt_aot_thread_has_started (ep_rt_thread_handle_t thread_handle)
+{
+    return thread_handle != NULL && thread_handle->IsInitialized ();
```

Comment:
How can you end up with a thread_handle that is not initialized here? The thread handles handed out by the methods above are always initialized.

Reply:
`ep_rt_aot_thread_get_handle `returns `GetCurrentThreadIfAvailable` which can return null.

Reply:
Right, that should be covered by `thread_handle != NULL` check. I agree the null check should be here. I am questing the need for `IsInitialized` part of the condition.